### DISCUSSION TITLE
Two static Csla.IDataPortalFactory instances

### DISCUSSION THREAD

**Discussion by michaelcsikos** [MarimerLLC/csla#4215] [Questions]
We have some common business classes in a shared DLL which are used by multiple add-ins for Advance Steel. The UI is WinForms. I'm thinking about a scenario where two different add-ins are loaded at the same time and both use some of the common classes, e.g. `Bolt` and `Plate`.

Using CSLA 8, both add-ins could set up their own `static Csla.IDataPortalFactory` and be able to handle the factory operations for the common classes. Does it actually matter which `IDataPortalFactory` instance handles the requests for the common classes?

**Comment by ossendorf-at-hoelscher** [comment]
Hi. In case of a WinForms app it should not matter because you typically use the root provider to retrieve all your dependencies.
The only case where that won't work - or at least might have some hard to find problems - would be when you use di scopes which are disposed during runtime. In that case you can for example hit ObjectDisposedExceptios because a BO resolved by such a scope uses the service provider of the scope.
Hth
